perceval/backends/stackexchange.py

```diff
@@ -0,0 +1,296 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Alberto Martín <alberto.martin@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import requests
+import time
+
+from ..backend import Backend, BackendCommand, metadata
+from ..cache import Cache
+from ..errors import CacheError
+from ..utils import str_to_datetime, DEFAULT_DATETIME, urljoin
+
+# Filters are immutable and non-expiring. This filter allows to retrieve all
+# the information regarding Each question. To know more, visit
+# https://api.stackexchange.com/docs/questions and paste the filter in the
+# whitebox filter. It will display a list of checkboxes with the selected
+# values for the filter provided.
+
+QUESTIONS_FILTER = 'Bf*y*ByQD_upZqozgU6lXL_62USGOoV3)MFNgiHqHpmO_Y-jHR'
+MAX_QUESTIONS = 100  # Maximum number of reviews per query
+STACKEXCHANGE_API_URL = 'https://api.stackexchange.com'
+VERSION_API = '2.2'
+
+logger = logging.getLogger(__name__)
+
+
+def get_update_time(item):
+    """Extracts the update time from a StackExchange item"""
+    return item['last_activity_date']
+
+
+class StackExchange(Backend):
+    """StackExchange backend for Perceval.
+
+    This class retrieves the questions stored in any of the
+    StackExchange sites. To initialize this class the
+    site and the tag must be provided.
+
+    :param site: StackExchange site
+    :param tagged: filter items by question Tag
+    :param token: StackExchange access_token for the API
+    :param cache: cache object to store raw data
+    """
+    version = '0.1.0'
+
+    def __init__(self, site=None, tagged=None, token=None,
+                 max_questions=None, cache=None):
+
+        super().__init__(site, cache=cache)
+        self.site = site
+        self.tagged = tagged
+        self.max_questions = max_questions
+        self.client = StackExchangeClient(site, tagged, token, max_questions)
+
+    @metadata(get_update_time)
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        """Fetch the questions from the site.
+
+        The method retrieves, from a StackExchange site, the
+        questions updated since the given date.
+
+        :param from_date: obtain questions updated since this date
+
+        :returns: a generator of questions
+        """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        logger.info("Looking for questions at site '%s', with tag '%s' and updated from '%s'",
+                    self.site, self.tagged, str(from_date))
+
+        self._purge_cache_queue()
+
+        whole_page = self.client.get_questions(from_date)
+
+        for questions in whole_page:
+            for question in questions:
+                self._push_cache_queue(question)
```

Comment:
What it should be stored in the cache is `whole_page` not `question`. This is a misconception of how we implemented backend clients so far. See my comments below on class `StackExchangeClient`.